cpg-language-configfiles/src/main/kotlin/de/fraunhofer/aisec/cpg/frontend/configfiles/IniFilesFrontend.kt

```diff
@@ -0,0 +1,128 @@
+ *   field and the [de.fraunhofer.aisec.cpg.graph.declarations.FieldDeclaration.initializer] is set
+ *   to a [Literal] with the corresponding `entry`s `value`.
+ *
+ * Note:
+ * - the "ini4j" library does not provide any super type for all nodes. Thus, the frontend accepts
+ *   `Any`
+ * - [typeOf] has to be implemented, but as there are no types always returns the builtin `string`
+ *   type
+ * - [codeOf] has to accept `Any` (because of the limitations stated above) and simply returns
+ *   `.toString()`
+ * - [locationOf] always returns `null` as the "ini4j" library does not provide any means of getting
+ *   a location given a node
+ * - [setComment] TODO what's this?
+ */
+class IniFilesFrontend(language: Language<IniFilesFrontend>, ctx: TranslationContext) :
+    LanguageFrontend<Any, Any?>(language, ctx) {
+
+    override fun parse(file: File): TranslationUnitDeclaration {
+        val ini = Ini()
+        try {
+            ini.load(FileInputStream(file))
+        } catch (ex: Exception) {
+            throw TranslationException("Parsing failed with exception: $ex")
+        }
+        val tud = newTranslationUnitDeclaration(name = file.name, rawNode = ini)
+        scopeManager.resetToGlobal(tud)
+        val nsd = newNamespaceDeclaration(name = file.name, rawNode = ini)
+        scopeManager.addDeclaration(nsd)
+        scopeManager.enterScope(nsd)
+
+        ini.values.forEach { handleSection(it) }
+
+        scopeManager.enterScope(nsd)
+        return tud
+    }
+
+    /**
+     * Translates a `Section` into a [RecordDeclaration] and handles all `entries` using
+     * [handleEntry].
+     */
+    private fun handleSection(section: Profile.Section) {
+        val record = newRecordDeclaration(name = section.name, kind = "section", rawNode = section)
+        scopeManager.addDeclaration(record)
+        scopeManager.enterScope(record)
+        section.entries.forEach { handleEntry(it) }
+        scopeManager.leaveScope(record)
+    }
+
+    /**
+     * Translates an `MutableEntry` to a new
+     * [de.fraunhofer.aisec.cpg.graph.declarations.FieldDeclaration] with the
+     * [de.fraunhofer.aisec.cpg.graph.declarations.FieldDeclaration.initializer] being set to the
+     * `entry`s value.
+     */
+    private fun handleEntry(entry: MutableMap.MutableEntry<String?, String?>) {
+        val field =
+            newFieldDeclaration(name = entry.key, type = primitiveType("string"), rawNode = entry)
+                .apply { initializer = newLiteral(value = entry.value, rawNode = entry) }
+        scopeManager.addDeclaration(field)
+    }
+
+    override fun typeOf(type: Any?): Type {
+        return primitiveType("string")
+    }
+
+    override fun codeOf(astNode: Any): String? {
+        return astNode.toString()
+    }
+
+    override fun locationOf(astNode: Any): PhysicalLocation? {
+        return null // currently, the line number / column cannot be accessed given an Ini object
```

Comment:
return entire file first line number to last line number

Reply:
I updated this to return a Region of -1, -1, -1, -1. Seems more reasonable to me.